runelite-client/src/main/java/net/runelite/client/plugins/shiftwalker/ShiftWalkerPlugin.java

```diff
@@ -0,0 +1,184 @@
+	private ConfigManager configManager;
+
+	@Inject
+	private KeyManager keyManager;
+
+	@Setter
+	private boolean hotKeyPressed = false;
+
+	@Provides
+	ShiftWalkerConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(ShiftWalkerConfig.class);
+	}
+
+	@Override
+	public void startUp()
+	{
+		keyManager.registerKeyListener(inputListener);
+	}
+
+	@Override
+	public void shutDown()
+	{
+		keyManager.unregisterKeyListener(inputListener);
+	}
+
+	@Subscribe
+	public void onFocusChanged(FocusChanged event)
+	{
+		if (!event.isFocused())
+		{
+			hotKeyPressed = false;
+		}
+	}
+
+	/**
+	 * Event when a new menu entry was added.
+	 * @param event {@link MenuEntryAdded}.
+	 */
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (client.getGameState() != GameState.LOGGED_IN || !hotKeyPressed)
+		{
+			return;
+		}
+
+		final String pOptionToReplace = Text.removeTags(event.getOption()).toUpperCase();
+
+		//If the option is already to walk there, or cancel we don't need to swap it with anything
+		if (pOptionToReplace.equals(CANCEL) || pOptionToReplace.equals(WALK_HERE))
+		{
+			return;
+		}
+
+		String target = Text.removeTags(event.getTarget().toUpperCase());
+
+		if (config.shiftWalkEverything())
+		{
+			swap(pOptionToReplace); //Swap everything with walk here
+		}
+		else if (config.shiftWalkBoxTraps() && ShiftWalkerGroups.BOX_TRAP_TARGETS.contains(target)
+				&& ShiftWalkerGroups.BOX_TRAP_KEYWORDS.contains(pOptionToReplace))
+		{
+			swap(pOptionToReplace); //Swap only on box traps
+		}
+		else if (config.shiftWalkAttackOption() && ShiftWalkerGroups.ATTACK_OPTIONS_KEYWORDS.contains(pOptionToReplace))
+		{
+			swap(pOptionToReplace); //Swap on everything that has an attack keyword as the first option
+		}
+	}
+
+	/**
+	 * Swaps menu entries if the entries could be found. This places Walk Here where the top level menu option was.
+	 * @param pOptionToReplace The String containing the Menu Option that needs to be replaced. IE: "Attack", "Chop Down".
+	 */
+	private void swap(String pOptionToReplace)
+	{
+		MenuEntry[] entries = client.getMenuEntries();
+
+		Integer walkHereEntry = searchIndex(entries, WALK_HERE);
+		Integer entryToReplace = searchIndex(entries, pOptionToReplace);
+
+		if (walkHereEntry != null
+			&& entryToReplace != null)
+		{
+			MenuEntry walkHereMenuEntry = entries[walkHereEntry];
+			entries[walkHereEntry] = entries[entryToReplace];
+			entries[entryToReplace] = walkHereMenuEntry;
+
+			client.setMenuEntries(entries);
+		}
+	}
+
+	/**
+	 * Finds the index of the menu that contains the verbiage we are looking for.
+	 * @param pMenuEntries The list of {@link MenuEntry}s.
+	 * @param pMenuEntryToSearchFor The Option in the menu to search for.
+	 * @return The index location or null if it was not found.
+	 */
+	private Integer searchIndex(MenuEntry[] pMenuEntries, String pMenuEntryToSearchFor)
+	{
+		Integer indexLocation = 0;
+
+		for (MenuEntry menuEntry : pMenuEntries)
```

Comment:
This needs to be a search in the reverse order than this. Searching like this only does the top level entry. But multiple entries in the menu won't get touched, meaning this won't work then.